### PR TITLE
Add metadata validation rule: dependencies

### PR DESCRIPTION
**Pull request by LowEntropyBody**
* Requires rule

### CODE REVIEW

**Comment by LowEntropyBody** [comment]
@kimizhu

File: test/docfx.Test/lib/JsonSchemaTest.cs [hunk 1]
```diff
@@ -188,6 +192,13 @@ public void TestJsonSchemaConfirmance(string description, string schemaText, str
         [InlineData("{'required': ['a']}", "{'a': 1}", "")]
         [InlineData("{'required': ['a']}", "{'b': 1}",
             "['warning','field-required','Missing required field 'a'','file',1,1]")]
+
+        // dependencies validation
+        [InlineData("{'dependencies': {}}", "{}", "")]
+        [InlineData("{'dependencies': {'key1': ['key2']}}", "{'key1' : 1, 'key2' : 2}", "")]
+        [InlineData("{'dependencies': {'key1': ['key2']}}", "{}", "")]
+        [InlineData("{'dependencies': {'key1': ['key2']}}", "{'key1' : 1}",
```
Comment (OsmondJiang) [inline test/docfx.Test/lib/JsonSchemaTest.cs]: would you please add non-root level dependencies test?

**Comment by herohua** [comment]
@LowEntropyBody : would you update the title of the PR to be more specific, e.g. something like 'metadata validation rule: dependencies'

**Comment by yufeih** [comment]
@LowEntropyBody , not related to the task, but if you like, you can also implement the schema form of  `dependencies` easily by using a tuple and `ValueOrObjectConverter`, this is [an example PR](https://github.com/dotnet/docfx/pull/4590/files#diff-164eae5cbda9e1df2421018a24192c99R44) for your reference

**Comment by LowEntropyBody** [comment]
Work item link: [100602](https://dev.azure.com/ceapex/Engineering/_workitems/edit/100602)

**Pull request merged**